core/domain/feature_gating_services_test.py

```diff
@@ -0,0 +1,217 @@
+                'app_version': '1.0.0',
+                'user_locale': 'en',
+            })
+            self.assertEqual(
+                feature_services.get_all_feature_flag_values_for_context(
+                    context),
+                {
+                    self.param_1.name: False,
+                    self.param_2.name: True,
+                }
+            )
+
+    def test_get_all_feature_flag_values_in_prod_returns_correct_values(self):
+        with self.swap(constants, 'DEV_MODE', False):
+            context = feature_services.create_evaluation_context_for_client({
+                'client_type': 'Android',
+                'browser_type': None,
+                'app_version': '1.0.0',
+                'user_locale': 'en',
+            })
+            self.assertEqual(
+                feature_services.get_all_feature_flag_values_for_context(
+                    context),
+                {
+                    self.param_1.name: False,
+                    self.param_2.name: False,
+                }
+            )
+
+    def test_get_feature_flag_value_in_dev_env_returns_correct_values(self):
+        with self.swap(constants, 'DEV_MODE', True):
+            self.assertFalse(
+                feature_services.get_feature_flag_value(self.param_1.name))
+            self.assertTrue(
+                feature_services.get_feature_flag_value(self.param_2.name))
+
+    def test_get_feature_flag_value_in_prod_env_returns_correct_values(self):
+        with self.swap(constants, 'DEV_MODE', False):
+            self.assertFalse(
+                feature_services.get_feature_flag_value(self.param_1.name))
+            self.assertFalse(
+                feature_services.get_feature_flag_value(self.param_2.name))
+
+    def test_get_feature_flag_values_with_invalid_name_failure(self):
+        with self.assertRaisesRegexp(Exception, 'not exist'):
+            feature_services.get_feature_flag_value(
+                'feature_that_does_not_exist')
+
+    def test_update_feature_flag_rules_successfully_updates_rules(self):
+        feature_services.update_feature_flag_rules(
+            self.param_1.name, self.user_id, 'test update',
+            [
+                {
+                    'filters': [
+                        {
+                            'type': 'server_mode',
+                            'conditions': [
+                                ['=', param_domain.FEATURE_STAGES.dev]
+                            ]
+                        }
+                    ],
+                    'value_when_matched': True
+                },
+            ]
+        )
+
+        with self.swap(constants, 'DEV_MODE', True):
+            self.assertTrue(
+                feature_services.get_feature_flag_value(self.param_1.name))
+            self.assertTrue(
+                feature_services.get_feature_flag_value(self.param_2.name))
+
+    def test_update_feature_flag_rules_with_invalid_name_failure(self):
```

Comment:
``test_update_feature_flag_rules_with_invalid_name_raises_error``

Ditto for the method below. I think this fits within 80 characters for both.